storm-core/src/clj/org/apache/storm/daemon/worker.clj

```diff
@@ -155,7 +155,10 @@
         ;; update the worker's backpressure flag to zookeeper only when it has changed
         (log-debug "BP " @(:backpressure worker) " WAS " prev-backpressure-flag)
         (when (not= prev-backpressure-flag @(:backpressure worker))
-          (.workerBackpressure storm-cluster-state storm-id assignment-id port @(:backpressure worker)))
+          (try
+            (.workerBackpressure storm-cluster-state storm-id assignment-id (long port) @(:backpressure worker))
```

Comment:
(:port worker) returns an Integer. Will cause an exception if not turning to long. See also Storm-1578.